source/PlayServicesResolver/src/ResolverVer1_1.cs

```diff
@@ -649,21 +651,29 @@ public override int Version()
                                                    gradleWrapper, gradleArgumentsString),
                                      level: LogLevel.Verbose);
 
-            // Run the build script to perform the resolution popping up a window in the editor.
-            var window = CommandLineDialog.CreateCommandLineDialog(
-                "Resolving Android Dependencies");
-            window.summaryText = "Resolving Android Dependencies....";
-            window.modal = false;
-            window.progressTitle = window.summaryText;
-            window.autoScrollToBottom = true;
-            window.RunAsync(gradleWrapper, gradleArgumentsString,
-                            (result) => {
-                                window.Close();
-                                scheduleOnMainThread(result);
-                            },
-                            workingDirectory: gradleBuildDirectory,
-                            maxProgressLines: 50);
-            window.Show();
+            if (runSynchronously)
+            {
+                CommandLine.Run(gradleWrapper, gradleArgumentsString, gradleBuildDirectory);
+            }
+            else
+            {
+                // Run the build script to perform the resolution popping up a window in the editor.
+                var window = CommandLineDialog.CreateCommandLineDialog(
+                    "Resolving Android Dependencies");
+                window.summaryText = "Resolving Android Dependencies....";
+                window.modal = false;
+                window.progressTitle = window.summaryText;
+                window.autoScrollToBottom = true;
+                window.RunAsync(gradleWrapper, gradleArgumentsString,
```

Comment:
CommandLineDialog.RunAsync() is a superset of CommandLine.Run() would it make sense to have a mode where CommandLineDialog.RunAsync() wouldn't pop up a window to run a command instead of branching here?  There are loads of places CommandLineDialog is used in the process beyond this code path.

Reply:
Not sure i understand. RunAsync starts a new thread, and invokes a callback when this thread is done. Run simple runs synchronously. 
Since i need to run synchronously, and RunAsync would be the wrong name if it in fact can run synchronously, then this doesn't seem to make much sense to me.

Reply:
Sorry about the slow response here.  I guess the point I was trying to make is that following the pattern that you're currently applying will end up duplicating all code paths that use CommandLineWindow.  To me this seems pretty unmaintainable.  If we want to support synchronous execution in this module then really everything should have the option of running synchronously.

Since it sounds like the problem you were originally trying to resolve was running resolution without the UI present (i.e Editor.update isn't called and windows don't work reliably) what I we can do instead is change CommandLineWindow to fall back to execution without a window and change the code which pumps the work queue (Editor.update) to instead have a general system for tasks run in the editor which can be configured to run everything synchronously if required.

Reply:
Ok, i see your point. But i have to say, this sounds like a bigger task than we are willing to put time into for the time being. We will just use our fork for now ;)